common/app/common/Preload.scala

```diff
@@ -1,15 +1,14 @@
 package common
 
 import model.ApplicationIdentity
-import conf.switches.Switches.PolyfillIO
 import html.HtmlPageHelpers.ContentCSSFile
 import play.api.mvc.RequestHeader
-import experiments.{ CommercialBaseline, ActiveExperiments}
+import experiments.{CommercialBaseline, ActiveExperiments}
 
 object Preload {
 
   def commercialBundleName(implicit request: RequestHeader): String =
-    if (CommercialBaseline.isParticipating(request, ActiveExperiments.canCheckExperiment)) {
```

Comment:
I'm just checking on CODE to confirm whether this fix will help. FYI @TBonnin , looks like Jon was unlucky enough to use the API which doesn't provide updates to the `ExperimentsFilter`

Reply:
Sorry I don't follow what do you mean by `doesn't provide updates to the ExperimentsFilter`

Reply:
The only way to ensure `ExperimentsDefinition.isIn()` and `LookedAtExperiments.addExperiment()` is to call `ExperimentsDefinition.isParticipating()`.

Jon was calling `Experiment.isParticipating()`, which does not update the hash map used by the `ExperimentsFilter`.

Reply:
Works now.

Reply:
Oh I see. The goal was to make calling `Experiment.isParticipating()` forbidden outside of `Experiment`. Looks like it is not the case at the moment and this should be fixed @nicl